server/src/main/java/org/candlepin/service/impl/DefaultEntitlementCertServiceAdapter.java

```diff
@@ -461,7 +461,7 @@ public String cleanUpPrefix(String contentPrefix) throws IOException {
                 ent.getQuantity(),
                 ent.getId());
 
-            Set<Product> products = new HashSet<>(productCurator.getPoolProvidedProductsCached(pool));
+            Set<Product> products = new HashSet<>(pool.getProduct().getProvidedProducts());
```

Comment:
I'm trying to find out if this change is enough or not. From what I can find out, the code in the `productCurator.getPoolProvidedProductsCached()` method was added in a0986b46f076be684fc742a761d4082ca916f437 as an attempt for better performance by caching provided products in JCache, and to use these methods on the curator instead of the `pool.getProvidedProducts/derivedProvidedProducts` getters. This was later changed by @barnabycourt to use Hibernate L2 cache in 404af5066f49f9aac350cfac86454e40d5f677c7.

My assumption is that the original mechanism with the productCurator methods is no longer needed, because Product is already immutable and cachable in hibernate L2, and the only thing we'd need to do is mark the `providedProducts` collection on Product with `@Cachable` and `@Cache(usage = CacheConcurrencyStrategy.READ_ONLY)`, just like the other collections, and caching will work implicitly (without using the productCurator workarounds).

@Ceiu @barnabycourt Are my assumptions correct, or am I missing something?

Reply:
That is my assumption as well

Reply:
@nikosmoum Added caching annotation on providedProducts and also, remove reference of `productCurator.getPoolProvidedProductsCached()` from other places. Only 3 more places where this method still being used and these will be updated in refresh PR.